src/features/api/lib/network-provider.ts

```diff
@@ -49,113 +49,115 @@ type NetworkProviderOptions = RequestInit & {
   headers?: Headers
 }
 
-class NetworkProvider {
-  private readonly options: NetworkProviderOptions
-
-  constructor(options: NetworkProviderOptions) {
-    this.options = options
-  }
-
-  static normalizeOptions(options: NetworkProviderOptions) {
-    if (!options.headers) {
-      options.headers = new Headers()
+/**
+ * Ensure minimal request options object.
+ */
+const normalizeOptions = (options: NetworkProviderOptions) => ({
+  ...options,
+  method: options.method ?? 'GET',
+  headers: options.headers ?? new Headers(),
+  interceptors: {
+    beforeRequest: options.interceptors?.beforeRequest ?? [],
+    afterRequest: options.interceptors?.afterRequest ?? [],
+  },
+})
```

Comment:
We use Extract Module technique to isolate methods and actions that can be tested without too much knowledge.